File: DesignPatterns/01_GOF23/DP_examples/CSharp/03_Behavioral/Command/Receiver.cs
namespace Behavioral.Command;

public record struct Stock(string Name, int Quantity);
public interface IReceiver
{
    void Action();
}
public class BuyStock(Stock stock) : IReceiver
{
    public void BuyStocks()
    {
        Console.WriteLine($"Buy {stock.Quantity} Stocks which name is {stock.Name}");
    }
    public void Action() => BuyStocks();
}
public class SellStock(Stock stock) : IReceiver
{
    public void SellStocks()
    {
        Console.WriteLine($"Sell {stock.Quantity} Stocks which name is {stock.Name}");
    }
    public void Action() => SellStocks();
}

File: DesignPatterns/01_GOF23/DP_examples/Python/03_Behavioral/Command/Receiver.py
# 接收者实现


class Stock:
    def __init__(self, name, quantity):
        self.name = name
        self.quantity = quantity


class IReceiver:
    def action(self):
        pass


class BuyStock(IReceiver):
    def __init__(self, stock):
        self.stock = stock
    
    def buy_stocks(self):
        print(f"Buy {self.stock.quantity} Stocks which name is {self.stock.name}")
    
    def action(self):
        self.buy_stocks()


class SellStock(IReceiver):
    def __init__(self, stock):
        self.stock = stock
    
    def sell_stocks(self):
        print(f"Sell {self.stock.quantity} Stocks which name is {self.stock.name}")
    
    def action(self):
        self.sell_stocks()
